dll/win32/shell32/dialogs/drive.cpp

```diff
@@ -159,123 +156,10 @@ GetDefaultClusterSize(LPWSTR szFs, PDWORD pClusterSize, PULARGE_INTEGER TotalNum
-    CoGetInterfaceAndReleaseStream(pPropData->pStream, IID_PPV_ARG(IDataObject, &pDataObj));
-
-    HPSXA hpsx = NULL;
-    HPROPSHEETPAGE hpsp[MAX_PROPERTY_SHEET_PAGE];
-    CComObject<CDrvDefExt> *pDrvDefExt = NULL;
-
-    CDataObjectHIDA cida(pDataObj);
-    if (FAILED_UNEXPECTEDLY(cida.hr()))
-        return FAILED(cida.hr());
-
-    RECT rcPosition = {CW_USEDEFAULT, CW_USEDEFAULT, 0, 0};
-    POINT pt;
-    if (SUCCEEDED(DataObject_GetOffset(pDataObj, &pt)))
-    {
-        rcPosition.left = pt.x;
-        rcPosition.top = pt.y;
-    }
-
-    DWORD style = WS_DISABLED | WS_CLIPSIBLINGS | WS_CAPTION;
-    DWORD exstyle = WS_EX_WINDOWEDGE | WS_EX_APPWINDOW;
-    CStubWindow32 stub;
-    if (!stub.Create(NULL, rcPosition, NULL, style, exstyle))
-    {
-        ERR("StubWindow32 creation failed\n");
-        return FALSE;
-    }
-
-    PROPSHEETHEADERW psh = {sizeof(PROPSHEETHEADERW)};
-    psh.dwFlags = PSH_PROPTITLE;
-    psh.pszCaption = pwszDrive;
-    psh.hwndParent = stub;
-    psh.nStartPage = 0;
-    psh.phpage = hpsp;
-
-    HRESULT hr = CComObject<CDrvDefExt>::CreateInstance(&pDrvDefExt);
-    if (SUCCEEDED(hr))
-    {
-        pDrvDefExt->AddRef(); // CreateInstance returns object with 0 ref count
-        hr = pDrvDefExt->Initialize(HIDA_GetPIDLFolder(cida), pDataObj, NULL);
-        if (SUCCEEDED(hr))
-        {
-            hr = pDrvDefExt->AddPages(AddPropSheetPageCallback, (LPARAM)&psh);
-            if (FAILED(hr))
-                ERR("AddPages failed\n");
-        }
-        else
-        {
-            ERR("Initialize failed\n");
-        }
-    }
-
-    hpsx = SHCreatePropSheetExtArrayEx(HKEY_CLASSES_ROOT, L"Drive", MAX_PROPERTY_SHEET_PAGE, pDataObj);
-    if (hpsx)
-        SHAddFromPropSheetExtArray(hpsx, (LPFNADDPROPSHEETPAGE)AddPropSheetPageCallback, (LPARAM)&psh);
-
-    INT_PTR ret = PropertySheetW(&psh);
-
-    if (hpsx)
-        SHDestroyPropSheetExtArray(hpsx);
-    if (pDrvDefExt)
-        pDrvDefExt->Release();
-
-    stub.DestroyWindow();
-
-    return ret != -1;
-}
-
 BOOL
 SH_ShowDriveProperties(WCHAR *pwszDrive, IDataObject *pDataObj)
 {
-    HRESULT hr = SHStrDupW(pwszDrive, &pwszDrive);
-    if (FAILED_UNEXPECTEDLY(hr))
-        return FALSE;
-
-    // Prepare data for thread
-    DRIVE_PROP_DATA *pData = (DRIVE_PROP_DATA *)SHAlloc(sizeof(*pData));
-    if (!pData)
-    {
-        SHFree(pwszDrive);
-        return FALSE;
-    }
-    pData->pwszDrive = pwszDrive;
-
-    // Marshall IDataObject to IStream
-    hr = CoMarshalInterThreadInterfaceInStream(IID_IDataObject, pDataObj, &pData->pStream);
-    if (SUCCEEDED(hr))
-    {
-        // Run a property sheet in another thread
-        if (SHCreateThread(ShowDrivePropThreadProc, pData, CTF_COINIT, NULL))
-            return TRUE; // Success
-
-        pData->pStream->Release();
-    }
-    SHFree(pData);
-    SHFree(pwszDrive);
-    return FALSE; // Failed
+    return SUCCEEDED(SHELL32_ShowFilesystemItemPropertiesDialogAsync(pDataObj));
```

Comment:
`pwszDrive` was used as the title of the property sheet dialog and this is incorrect because the title is supposed to be the display name of the drive (which might include a label etc.).